meson.build

```diff
@@ -0,0 +1,32 @@
+project(
+    'ddbus',
+    'd',
+    meson_version: '>=0.50',
+    version: '3.0.0',
+)
+
+src_inc = include_directories('source')
+
+prefix = get_option('prefix')
+include_dir = join_paths(prefix, get_option('includedir'), 'd', 'apk-polkit')
+
+vcs_tag = run_command('git', 'rev-parse', '--short', 'HEAD').stdout().strip()
+if vcs_tag == ''
+    corecollector_version = meson.project_version()
+else
+    corecollector_version = meson.project_version() + '-@0@'.format (vcs_tag)
+endif
+
+compiler_id = meson.get_compiler('d').get_id()
+
+link_runtime_shared_arg = []
+
+if compiler_id == 'llvm'
+    link_runtime_shared_arg = ['-link-defaultlib-shared']
+elif compiler_id == 'dmd'
+    link_runtime_shared_arg = ['-defaultlib=phobos2', '-debuglib=phobos2']
+elif compiler_id == 'gcc'
+    link_runtime_shared_arg = ['-shared-libphobos']
+endif
```

Comment:
This is \*still\* wrong. Please stop ignoring my comments.

Reply:
At least with gdc this is required, without this:

```
ldd build/source/ddbus/libddbus.so
	/lib/ld-musl-x86_64.so.1 (0x7f99671f8000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x7f9966f4c000)
	libc.musl-x86_64.so.1 => /lib/ld-musl-x86_64.so.1 (0x7f99671f8000)
```

With this:
```
ldd build/source/ddbus/libddbus.so
	/lib/ld-musl-x86_64.so.1 (0x7f8679513000)
	libgphobos.so.76 => /usr/lib/libgphobos.so.76 (0x7f8678f78000)
	libgdruntime.so.76 => /usr/lib/libgdruntime.so.76 (0x7f8678e49000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x7f8678e35000)
	libc.musl-x86_64.so.1 => /lib/ld-musl-x86_64.so.1 (0x7f8679513000)
```

Reply:
No. That uses the default behavior for your istallation. If you want to overwrite it you can pass do `DFLAGS=-shared-libphobos`. Patching the downstream like this is plain wrong. You are forcing a build flag when the build system is perfectly capable of adding it if the user chooses so.

Reply:
Example of how this behavior is wrong:
`libphobos` is not built as a shared library by default. Forcing `-shared-libphobos` means that users with such installation can't build the project.

If your system doesn't have the default behavior you desire, you should change it. If the installation is provided by your distribution then request them to change the default behavior, if you are using your own toolchain, build it with the correct flags.

Reply:
There is no situation where you'd want a build with a static phobos though, since linking a shared library with a static phobos will blow up once you link it against an executable. I guess I'll remove the workaround though...

Reply:
There are situations where you might want to do that. Nevertheless, we should respect the default, a downstream should never be this opinionated.

Reply:
Fair, removed it.